tests/by-util/test_sort.rs

```diff
@@ -1568,3 +1568,177 @@ fn test_g_float_hex() {
+}
+
+#[test]
+fn test_clap_localization_help_message() {
+    // Test help message in English
+    let result_en = new_ucmd!()
+        .env("LANG", "en_US.UTF-8")
+        .env("LC_ALL", "en_US.UTF-8")
+        .arg("--help")
+        .succeeds();
+
+    let stdout_en = result_en.stdout_str();
+    assert!(stdout_en.contains("Usage:"));
+    assert!(stdout_en.contains("Options:"));
+
+    // Test help message in French
+    let result_fr = new_ucmd!()
+        .env("LANG", "fr_FR.UTF-8")
+        .env("LC_ALL", "fr_FR.UTF-8")
+        .arg("--help")
+        .succeeds();
+
+    let stdout_fr = result_fr.stdout_str();
+    assert!(stdout_fr.contains("Utilisation:"));
+    assert!(stdout_fr.contains("Options:"));
+}
+
+#[test]
+fn test_clap_localization_version() {
+    // Test version output (should exit with 0)
+    let result = new_ucmd!().arg("--version").succeeds();
+
+    let stdout = result.stdout_str();
+    assert!(stdout.contains("sort"));
+}
+
+#[test]
+fn test_clap_localization_missing_required_argument() {
+    // Test missing required argument
+    let result_en = new_ucmd!()
+        .env("LANG", "en_US.UTF-8")
+        .env("LC_ALL", "en_US.UTF-8")
+        .arg("-k")
+        .fails();
+
+    let stderr_en = result_en.stderr_str();
+    assert!(stderr_en.contains(" a value is required for '--key <key>' but none was supplied"));
+    assert!(stderr_en.contains("-k"));
+}
+
+#[test]
+fn test_clap_localization_invalid_value() {
+    // Test invalid value error
+    let result_en = new_ucmd!()
+        .env("LANG", "en_US.UTF-8")
+        .env("LC_ALL", "en_US.UTF-8")
+        .arg("-k")
+        .arg("invalid")
+        .fails();
+
+    let stderr_en = result_en.stderr_str();
+    assert!(stderr_en.contains("sort: failed to parse key 'invalid'"));
+
+    // Test in French
+    let result_fr = new_ucmd!()
+        .env("LANG", "fr_FR.UTF-8")
+        .env("LC_ALL", "fr_FR.UTF-8")
+        .arg("-k")
+        .arg("invalid")
+        .fails();
+
+    let stderr_fr = result_fr.stderr_str();
+    assert!(stderr_fr.contains("sort: échec d'analyse de la clé 'invalid'"));
+}
+
+#[test]
+fn test_clap_localization_tip_for_value_with_dash() {
+    // Test tip for passing values that look like options
+    let result_en = new_ucmd!()
+        .env("LANG", "en_US.UTF-8")
+        .env("LC_ALL", "en_US.UTF-8")
+        .arg("--output")
+        .arg("--file-with-dash")
+        .fails();
+
+    let stderr_en = result_en.stderr_str();
+    assert!(stderr_en.contains("tip:") || stderr_en.contains("conseil:"));
+    assert!(stderr_en.contains("-- --file-with-dash"));
+
+    // Test in French
+    let result_fr = new_ucmd!()
+        .env("LANG", "fr_FR.UTF-8")
+        .env("LC_ALL", "fr_FR.UTF-8")
+        .arg("--output")
+        .arg("--file-with-dash")
+        .fails();
+
+    let stderr_fr = result_fr.stderr_str();
+    // The tip should be preserved from clap
+    assert!(stderr_fr.contains("tip:") || stderr_fr.contains("conseil:"));
```

Comment:
I don't understand why "tip" is not translated and why it is still checked for the French version.